infer/function.go

```diff
@@ -154,12 +167,7 @@ func (r *derivedInvokeController[F, I, O]) Invoke(ctx context.Context, req p.Inv
 		return p.InvokeResponse{}, fmt.Errorf("unable to apply defaults: %w", err)
 	}
 
-	var f F
-	// If F is a *struct, we need to rehydrate the underlying struct
-	if v := reflect.ValueOf(f); v.Kind() == reflect.Pointer && v.IsNil() {
-		f = reflect.New(v.Type().Elem()).Interface().(F)
-	}
-	o, err := f.Invoke(ctx, FunctionRequest[I]{Input: i})
+	o, err := r.receiver.Invoke(ctx, FunctionRequest[I]{Input: i})
```

Comment:
You can see how this also starts to help make it easier to reason about the framework.